azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java

```diff
@@ -5,34 +5,176 @@
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 public class Account {
     @Getter
-    private boolean authenticated;
+    @Setter
+    private AccountEntity entity;
+
     @Setter
     @Getter
-    private AzureEnvironment environment;
-    @Getter
-    private List<SubscriptionEntity> selectedSubscriptions;
+    private ICredentialProvider credentialBuilder;
 
-    private List<SubscriptionEntity> subscriptions;
+    private Map<String, TokenCredential> tenantToCredential = new HashMap<>();
 
     public Account logout() {
-        this.authenticated = false;
+        this.entity = null;
+        this.credentialBuilder = null;
+        this.tenantToCredential = new HashMap<>();
         return this;
     }
 
-    /**
-     * @return the credential for specified subscription
-     */
-    public AzureTokenCredentials getCredential(String subscriptionId) {
+    public AzureEnvironment getEnvironment() {
+        return entity.getEnvironment();
+    }
+
+    public void initialize() {
+        Objects.requireNonNull(entity, "Cannot initialize from null account entity.");
+        AzureEnvironment env = MoreObjects.firstNonNull(getEnvironment(), AzureEnvironment.AZURE);
+        AzureProfile azureProfile = new AzureProfile(env);
+        Set<String> validTenantIds = new HashSet<>();
+        if (this.entity.getTenantIds() == null) {
+            this.entity.setTenantIds(AzureResourceManager.authenticate(this.credentialBuilder.provideCredentialCommon()
+                    , azureProfile).tenants().list().stream().map(Tenant::tenantId).collect(Collectors.toList()));
+
+        }
+
+        Map<String, SubscriptionEntity> subscriptionMap = new HashMap<>();
+        this.entity.getTenantIds().forEach(tenantId -> {
+            try {
+                List<SubscriptionEntity> subscriptionsOnTenant =
+                        AzureResourceManager.authenticate(this.credentialBuilder.provideCredentialForTenant(tenantId), azureProfile).subscriptions().list()
+                                .mapPage(s -> this.toSubscriptionEntity(entity.getEnvironment(), tenantId, s)).stream().collect(Collectors.toList());
+
+                for (SubscriptionEntity s : subscriptionsOnTenant) {
+                    String key = StringUtils.lowerCase(s.getId());
+                    if (subscriptionMap.putIfAbsent(key, s) != null) {
+                        validTenantIds.add(tenantId);
+                    }
+                }
+            } catch (Exception ex) {
+                // ignore AuthenticationException since on some tenants, it doesn't allow list subscriptions
+                if (!(ExceptionUtils.getRootCause(ex) instanceof AuthenticationException)) {
```

Comment:
for other exception, do we need to throw them out?

Reply:
Let it throw exception and when user reports a valid exception, we should allow it here.